app/scripts/modules/core/src/presentation/hooks/useDimensions.hook.ts

```diff
@@ -0,0 +1,33 @@
+import { debounce } from 'lodash';
+import React from 'react';
+
+const getElementDimensions = (ref: React.RefObject<HTMLElement>) =>
+  ref.current ? { width: ref.current.offsetWidth, height: ref.current.offsetHeight } : { width: 0, height: 0 };
+
+export const useElementDimensions = ({
+  ref,
```

Comment:
this is just a style preference comment, not feedback per se

When an arg is "required" I prefer to pass it as a parameter, not as part of an object.  i.e., I would have probably written this as:

```
export const useElementDimension = (ref: ..., options: ...) => 
```